scene/3d/collision_shape.cpp

```diff
@@ -228,7 +228,9 @@ void CollisionShape::_update_debug_shape() {
 
 void CollisionShape::_shape_changed() {
 	// If this is a heightfield shape our center may have changed
-	_update_in_shape_owner(true);
+	if (parent) {
```

Comment:
That test should be done in `_update_in_shape_owner(true)` itself, with
```
if (!parent)
	return;
```

Reply:
I second that :)

Reply:
I was going to do this, but I felt as if it was unnecessary to check every time as most of the calling code already checks. Maybe a comment to specify that it is necessary?

Reply:
That's a good point, all other calls to this method ensure `if (parent)` before calling it.
IMO they could all be simplified by putting the call in that method and removing the `if (parent)` on calling sites, but that could be done in another PR, this one is consistent with the current codebase.